crates/wasmcloud-provider-macros/src/lib.rs

```diff
@@ -0,0 +1,371 @@
+use proc_macro2::TokenTree;
+
+#[proc_macro]
+pub fn generate(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
+    // Run wit-bindgen generation for the interface
+    let item: proc_macro2::TokenStream = input.into();
+
+    // Ensure that we have the args we expect
+    let tokens = item.clone().into_iter().collect::<Vec<TokenTree>>();
+    let (impl_struct_name, rest) = match tokens.split_first() {
+        Some((TokenTree::Ident(name), rest)) => (name, rest),
+        structure => {
+            eprintln!("structure {:?}", structure);
+            panic!(
+                r#"
+unexpected macro token structure. Please provide the name of your impl struct first and arguments to wit-bindgen afterwards. (ex. 'wit_bindgen_wasmcloud::generate!(MyActor, "path/to/types.wit"))
+"#
+            );
+        }
+    };
+
+    // Seperate the wit bindgen args
+    let mut bindgen_args = proc_macro2::TokenStream::new();
+    bindgen_args.extend(Vec::from(rest));
+
+    let ts = quote::quote!(
+        use wit_bindgen_wasmcloud::wasmtime as wasmtime;
+        wasmtime::component::bindgen!(#bindgen_args);
+
+        // use wit_bindgen_wasmcloud::serde::{Serialize, Deserialize};
+        use ::serde::{Serialize, Deserialize};
+        use wit_bindgen_wasmcloud::async_trait::async_trait;
```

Comment:
So this is *on purpose* -- so we can encapsulate all the extra deps in `wit_bindgen_wasmcloud`, rather than forcing the downstream to pull all of them in